client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -377,6 +378,12 @@ function mediaButton( editor ) {
 				editor.fire( 'SetTextAreaContent', { content } );
 			}
 
+			// We have just replaced some image nodes, potentially losing selection/focus.
+			// So if an image node was selected and one isn't now, re-select it.
```

Comment:
thanks for the comment update. I think it gets us closer but not fully to our ideal. if I read this I would have no idea that a duplication bug existed without it.

is duplicating the image directly related to losing selection? what would happen if we didn't reselect?